src/main/kotlin/lotto/domain/LottoTicket.kt

```diff
@@ -1,34 +1,36 @@
 package lotto.domain
 
-class LottoTicket(lottoNumbers: List<LottoNumber>) {
-    val value: List<LottoNumber> = lottoNumbers.sortedBy { it.value }
+import java.util.TreeSet
 
-    fun getCountOfMatch(winningNumbers: List<LottoNumber>): Int {
-        return value.count { lottoNumber ->
-            winningNumbers.contains(lottoNumber)
-        }
-    }
+class LottoTicket(val value: TreeSet<LottoNumber>) {
 
-    companion object {
-        private const val MIN_LOTTO_NUMBER = 1
-        private const val MAX_LOTTO_NUMBER = 45
+    init {
+        checkValidateLottoTicket()
+    }
 
-        fun generateAuto(): LottoTicket {
-            val numbers = (MIN_LOTTO_NUMBER..MAX_LOTTO_NUMBER).shuffled().slice(0..5)
+    constructor(numbers: List<Int>) : this(
+        TreeSet(
+            numbers.map { number ->
+                LottoNumber.from(number)
+            }
+        )
+    )
 
-            return LottoTicket(
-                numbers.map { number ->
-                    LottoNumber(number)
-                }
-            )
+    fun getCountOfMatch(lottoTicket: LottoTicket): Int {
+        return lottoTicket.value.count { lottoNumber ->
+            value.contains(lottoNumber)
         }
```

Comment:
아래의 코드로 개선해 보면 어떨까요?

```suggestion
        return lottoTicket.value.count(::isNumberContains)
```